src/main/scala/com/jakainas/functions/package.scala

```diff
@@ -65,4 +65,32 @@ package object functions {
       else deduped.as[T](conv)
     }
   }
+
+  implicit class DataFrameFunctions[T](val df: DataFrame) {
```

Comment:
`implicit class DataFrameFunctions[T](val df: DataFrame) extends AnyVal` to get a performance boost (AnyVal will allow scala to erase the class object creation and rewrite them as function calls)

Reply:
Thanks @kanielc . 

I see you moved function to `implicit cast DatasetFunctions[T]` -- does this alteration have similar performance boost? If so how given it not seem to have "extends AnyVal"? 

How did you accumulate this nugget of info re: performance boost? Experience? Some rockin' and informative article? Something uncovered using a scala perf testing framework of some sort?

Reply:
https://docs.scala-lang.org/overviews/core/value-classes.html

```
The following fragment of RichInt shows how it extends Int to allow the expression 3.toHexString:

implicit class RichInt(val self: Int) extends AnyVal {
  def toHexString: String = java.lang.Integer.toHexString(self)
}
At runtime, this expression 3.toHexString is optimised to the equivalent of a method call on a static object (RichInt$.MODULE$.extension$toHexString(3)), rather than a method call on a newly instantiated object.
```